### ISSUE TITLE
Single-indent multi-line function definitions

### ISSUE THREAD

**Issue by d-morrison** [jimhester/lintr#2830]
The "Good" "Single-indent" example of "Multi-line function definitions" from the tidyverse style guide (https://style.tidyverse.org/functions.html#multi-line-function-definitions) currently generates a lint from `indentation_linter`:

``` r
code <-
'long_function_name <- function(
    a = "a long argument",
    b = "another argument",
    c = "another long argument"
) {
  # As usual code is indented by two spaces.
}'
cat(code)
#> long_function_name <- function(
#>     a = "a long argument",
#>     b = "another argument",
#>     c = "another long argument"
#> ) {
#>   # As usual code is indented by two spaces.
#> }
lintr::lint(code)
#> <text>:2:4: style: [indentation_linter] Indentation should be 2 spaces but is 4 spaces.
#>     a = "a long argument",
#>   ~^
```

<sup>Created on 2025-03-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

This occurs both with v3.2.0 from CRAN and commit https://github.com/r-lib/lintr/commit/a2b8d6fec967e037e736ba81eb7fca83333438c7 from GitHub.

In contrast, moving `) {` onto the line of the last argument produces no lint:

``` r
code <-
'long_function_name <- function(
    a = "a long argument",
    b = "another argument",
    c = "another long argument") {
  # As usual code is indented by two spaces.
}'
cat(code)
#> long_function_name <- function(
#>     a = "a long argument",
#>     b = "another argument",
#>     c = "another long argument") {
#>   # As usual code is indented by two spaces.
#> }
lintr::lint(code)
#> ℹ No lints found.
```

<sup>Created on 2025-03-10 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
Double indentation for argument lists was removed from the tidyverse guide. So to me this is expected default behavior?

**Comment by d-morrison** [comment]
Ah - I didn't realize that those indents had gotten doubled when I set up the reprex (they are un-doubled in the style guide). 
I think "Reindent lines" (Ctrl-I) in RStudio still adds the double-indent (c.f. https://github.com/rstudio/rstudio/issues/10752 and https://github.com/rstudio/rstudio/issues/11101).

If double-indentation has been removed from the style guide, should the second example above also throw a lint? It also has double indentation, but currently no lint is thrown. Here's a slightly clearer version:

``` r
code <-
'
long_function_name <- function(
    a = "a long argument",
    b = "another argument",
    c = "another long argument") {
  # As usual code is indented by two spaces.
}
'
cat(code)
#> 
#> long_function_name <- function(
#>     a = "a long argument",
#>     b = "another argument",
#>     c = "another long argument") {
#>   # As usual code is indented by two spaces.
#> }
lintr::lint(code)
#> ℹ No lints found.
```

<sup>Created on 2025-03-11 with [reprex v2.1.1](https://reprex.tidyverse.org)</sup>

**Comment by AshesITR** [comment]
Yes I agree that could lint.
IMO it's not an indentation lint though, I think the lint should land on `)` which should be on a new line since the corresponding `(` is last-on-line.

WDYT, @MichaelChirico and @IndrajeetPatil?
